assets/js/base/components/payment-methods/saved-payment-method-options.js

```diff
@@ -92,63 +93,53 @@ const SavedPaymentMethodOptions = ( { onSelect } ) => {
 		setActivePaymentMethod,
 	} = usePaymentMethodDataContext();
 	const [ selectedToken, setSelectedToken ] = useState( '' );
+	const standardMethods = getPaymentMethods();
 
 	/**
 	 * @type      {Object} Options
 	 * @property  {Array}  current  The current options on the type.
 	 */
 	const currentOptions = useRef( [] );
 	useEffect( () => {
-		let options = [];
-		const paymentMethodKeys = Object.keys( customerPaymentMethods );
-		if ( paymentMethodKeys.length > 0 ) {
-			paymentMethodKeys.forEach( ( type ) => {
-				const paymentMethods = customerPaymentMethods[ type ];
-				if ( paymentMethods.length > 0 ) {
-					options = options.concat(
-						paymentMethods.map( ( paymentMethod ) => {
-							const option =
-								type === 'cc' || type === 'echeck'
-									? getCcOrEcheckPaymentMethodOption(
-											paymentMethod,
-											setActivePaymentMethod,
-											setPaymentStatus
-									  )
-									: getDefaultPaymentMethodOptions(
-											paymentMethod,
-											setActivePaymentMethod,
-											setPaymentStatus
-									  );
-							if (
-								paymentMethod.is_default &&
-								selectedToken === ''
-							) {
-								setSelectedToken( paymentMethod.tokenId + '' );
-								option.onChange( paymentMethod.tokenId );
-							}
-							return option;
-						} )
-					);
-				}
-			} );
-			if ( options.length > 0 ) {
-				currentOptions.current = options;
-				currentOptions.current.push( {
-					value: '0',
-					label: __(
-						'Use a new payment method',
-						'woo-gutenberg-product-blocks'
-					),
-					name: `wc-saved-payment-method-token-new`,
+		const types = Object.keys( customerPaymentMethods );
+		const options = types
+			.flatMap( ( type ) => {
+				const typeMethods = customerPaymentMethods[ type ];
+				return typeMethods.map( ( paymentMethod ) => {
+					const method =
+						standardMethods[ paymentMethod.method.gateway ];
+					if ( ! method?.supports?.savePaymentInfo ) {
```

Comment:
This block of code might be easier to review with `Hide whitespace changes`. The only change inside this `useEffect` which is not a refactor is this new check: if `savePaymentInfo` is not truthy, the saved payment methods are not appended (ie: if Stripe fails to load, saved credit cards will not be appended).

Before, that would be achieved because the payment method would not have been registered at all. But with the changes in this PR, all payment methods are registered for admins, so we need to add this extra check.